src/core/lib/layer.js

```diff
@@ -535,72 +506,47 @@ export default class Layer {
       propsOrDataChanged,
       viewportChanged,
       somethingChanged,
-      reason: dataChangedReason || propsChangedReason || 'Viewport changed'
+      reason: dataChanged || propsChanged || 'Viewport changed'
```

Comment:
Are we assuming that viewport is always changed?

Reply:
Should also use `updateTriggersChanged`?

Reply:
> Are we assuming that viewport is always changed?
> Should also use updateTriggersChanged?

Fixed. This code will be removed in a followup diff but I made an update that makes sure this PR leaves things in a good state, in case the next PR gets held up in review.